en/references/tested-judge/template-description/README.md

```diff
@@ -0,0 +1,392 @@
+  
+## Constants
+First we will look to the constants that could be used in the templates.
+- **prompt**:
+  This constant contains the console-prompt-prefix.
+  In the HTML template it will be the escaped version.
+  Example: `>>>` for Python.
+- **language and language_html**:
+  These constants contains the name of the programming language, wherefore an instance is generated.
+  `language_html` is the HTML-safe version of `language`.
+- **namespace and namespace_html**:
+  These constants contains the `namespace` of the code.
+  The namespace could be passed as a parameter to the conversion program.
+  `namespace_html` is the HTML-safe version of `namespace`.
+
+The constants and function calls in Mako must be notated between `${...}`.
+
+### Example
+The first example is in Markdown for Java and Python, with the namespace `solution`.
+The second example is in HTML for JavaScript, with the namespace `submission`.
+
+#### Markdown
+Template:
+```mako
+The prompt for **${language}** is `${prompt}`.
+The namespace is '${namespace}'.
+```
+
+Instance Java:
+```markdown
+The prompt for **java** is `>`.
+The namespace is 'Solution'.
+```
+
+Instance Python:
+```markdown
+The prompt for **python** is `>>>`.
+The namespace is 'solution'.
+```
+
+#### HTML
+Template:
+```mako
+<p>
+    The prompt for <span style="font-weight: bold">${language_html}</span> is <code>${prompt}</code>.
+    The namespace is '${namespace_html}'.
+</p>
+```
+
+Instance JavaScript:
+```html
+<p>
+    The prompt for <span style="font-weight: bold">javascript</span> is <code>&gt;</code>.
+    The namespace is 'submission'.
+</p>
+```
+
+## Functions
+As already indicated in the section [Constants](#constants) Mako also supports functions.
+
+We will again given an overview of all available functions that could be used in the templates with some examples.
+
+- **function_name**:
+  This function is used the generated the correct style conventions for the function names.
+  The input of this function is one string.
+  ::: tip Hint
+  When you pass the function name in the template, it's best to use the `snake_case` style convention.
+  This makes it possible to generate the correct function names foreach language.
+  :::
+- **type_name**:
+  This function is used to translate the TESTed datatypes to the programming language datatypes.
+  The input of this function is either a string, or a pair as type argument.  
+  The first argument of the pair must be a collection datatype string,
+  the second either one type argument, or a list of type arguments.
+- **natural_type_name**:
+  This function searches the natural name for a collectiondatatype and/or the string-datatype of TESTed for a given
+  programming language.
+  The input of this function is one string.
+- **statement and expression**:
+  This functions are not meant to use by the redactor of the descriptions.
+  But for the sake of completeness, they are listed anyway.
```

Comment:
If not meant to be used, we shouldn't mention them, since people will use them anyway, regardless of the warning.